publisher/publisher.go

```diff
@@ -149,6 +150,54 @@ func dropNegativeTimes(ev *event.Event) {
 	}
 }
 
+// parse the included X-Amzn-Trace-Id header if it is present in an ALB access
+// log - see
+// https://docs.aws.amazon.com/elasticloadbalancing/latest/application/load-balancer-request-tracing.html
+// for reference
+func addTraceData(ev *event.Event) {
+	// in the original access log field the header is called 'trace_id'
+	amznTraceID, ok := ev.Data["trace_id"].(string)
+	if !ok {
+		return
+	}
+	fields := strings.Split(amznTraceID, ";")
+	rootSpan := true
+	for _, field := range fields {
+		kv := strings.Split(field, "=")
+		key := kv[0]
+		val := kv[1]
+		switch key {
+		case "Root":
+			versionTimeID := strings.Split(val, "-")
+			ev.Data["trace.trace_id"] = versionTimeID[2]
+		case "Self":
+			versionTimeID := strings.Split(val, "-")
+			ev.Data["trace.span_id"] = versionTimeID[2]
+			rootSpan = false
+		case "Parent":
```

Comment:
I don't see mention of `Parent` in the AWS docs. Can you say more here?

Reply:
It's used in an [example in the X-Ray docs](https://docs.aws.amazon.com/xray/latest/devguide/xray-concepts.html#xray-concepts-tracingheader).

Reply:
Yea unfortunately I think making full use of this will be more complicated than just forwarding the headers directly.